R/driver-oracle.R

```diff
@@ -122,7 +122,12 @@ setMethod("odbcConnectionColumns_", c("Oracle", "character"),
       )
     }
 
-    dbGetQuery(conn, query)
+    res <- dbGetQuery(conn, query)
+
+    res$data_type <- as.numeric(res$data_type)
+    res[res$`field.type` == "DATE", c("data_type", "column_size")] <- c(91, 6)
+    res[grepl("TIMESTAMP", res$`field.type`), c("data_type", "column_size")] <- c(93, 16)
```

Comment:
```suggestion
    res[res$field.type == "DATE", c("data_type", "column_size")] <- c(91, 6)
    res[grepl("TIMESTAMP", res$field.type), c("data_type", "column_size")] <- c(93, 16)
```

No need for backticks with a `.` :)

``` r
x <- data.frame(dotted.column = 1)
x$dotted.column
#> [1] 1
```

<sup>Created on 2024-05-28 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

Reply:
Could we also drop an explanatory comment above these? e.g.

```suggestion
    # Overwrite type and precision/size using ODBC standard codes (#349, #350, #391).
    res[res$field.type == "DATE", c("data_type", "column_size")] <- c(91, 6)
    res[grepl("TIMESTAMP", res$field.type), c("data_type", "column_size")] <- c(93, 16)
```

Reply:
And maybe a link to where the field type numbers come from?

Reply:
Done, thanks